src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -110,14 +114,68 @@ private IntPtr AllocException(Exception ex)
         [DllImport(JitSupportLibrary)]
         private extern static char* GetExceptionMessage(IntPtr obj);
 
-        public static void Startup()
+
+        public static void Startup(TargetOS targetOS, TargetArchitecture targetArchitecture)
         {
-            jitStartup(GetJitHost(JitConfigProvider.Instance.UnmanagedInstance));
+            Debug.Assert(s_jitLibrary == IntPtr.Zero);
+
+            string crossgenPath = Path.GetDirectoryName(typeof(CorInfoImpl).Assembly.Location);
+            string jitTargetPath = Path.Combine(crossgenPath, GetTargetSpec(targetOS, targetArchitecture));
+            string jitLibraryName = Path.Combine(jitTargetPath, GetLibraryPrefix() + JitLibrary + GetLibraryExtension());
+
+            s_jitLibrary = NativeLibrary.Load(jitLibraryName);
+            s_jitStartup = Marshal.GetDelegateForFunctionPointer<JitStartupDelegate>(NativeLibrary.GetExport(s_jitLibrary, "jitStartup"));
+            s_getJit = Marshal.GetDelegateForFunctionPointer<GetJitDelegate>(NativeLibrary.GetExport(s_jitLibrary, "getJit"));
+
+            s_jitStartup(GetJitHost(JitConfigProvider.Instance.UnmanagedInstance));
+        }
+
+        private static string GetLibraryPrefix()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "" : "lib";
+        }
+
+        private static string GetLibraryExtension()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return ".dll";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return ".so";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return ".dylib";
+            }
+            else
+            {
+                throw new NotImplementedException("GetLibraryExtension");
+            }
+        }
+
+        private static string GetTargetSpec(TargetOS targetOS, TargetArchitecture targetArchitecture)
+        {
+            string targetOSComponent = targetOS switch
+            {
+                TargetOS.Windows => "win",
+                TargetOS.Linux => "lnx",
```

Comment:
Nit: I would use `linux` same as what we are using e.g. for RIDs. I do not think we are using "lnx" as the abbreviation for Linux anywhere else.

Reply:
Fixed in 19th commit, thanks for pointing that out.